src/lib/InstantSearch.ts

```diff
@@ -256,6 +262,13 @@ See ${createDocumentationLink({
       const newMiddleware = fn({ instantSearchInstance: this });
       this.middleware.push(newMiddleware);
 
+      if (
+        !this._insightsMiddleware &&
+        isInsightsMiddlewareDefinition(newMiddleware)
+      ) {
+        this._insightsMiddleware = newMiddleware;
+      }
```

Comment:
I think this code should be _inside_ the middleware

Reply:
done https://github.com/algolia/instantsearch.js/pull/4456/commits/19751711777f8621e483d5d4a33ada728f938ae6